src/stream/src/executor/hash_join.rs

```diff
@@ -682,18 +686,22 @@ impl<K: HashKey, S: StateStore, const T: JoinTypePrimitive> HashJoinExecutor<K,
 
                     // Report metrics of cached join rows/entries
                     for (side, ht) in [("left", &self.side_l.ht), ("right", &self.side_r.ht)] {
-                        self.metrics
-                            .join_cached_rows
-                            .with_label_values(&[&actor_id_str, side])
-                            .set(ht.cached_rows() as i64);
+                        // TODO(yuhao): Those two metric calculation cost too much time (>250ms).
+                        // Those will result in that barrier is always ready
+                        // in source. Since select barrier is preferred,
+                        // chunk would never be selected.
+                        // self.metrics
+                        //     .join_cached_rows
+                        //     .with_label_values(&[&actor_id_str, side])
+                        //     .set(ht.cached_rows() as i64);
                         self.metrics
                             .join_cached_entries
                             .with_label_values(&[&actor_id_str, side])
                             .set(ht.entry_count() as i64);
-                        self.metrics
-                            .join_cached_estimated_size
-                            .with_label_values(&[&actor_id_str, side])
-                            .set(ht.estimated_size() as i64);
+                        // self.metrics
+                        //     .join_cached_estimated_size
+                        //     .with_label_values(&[&actor_id_str, side])
+                        //     .set(ht.estimated_size() as i64);
```

Comment:
because it uses an `iter` to calculate the total size from scratch every time instead of maintaining the size incrementally? can we change it to the latter approach

Reply:
Yes! I'll do it in later pr.

Reply:
#5597